src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_HostObjectMemberReference.cs

```diff
@@ -12,7 +10,7 @@ internal partial class LocalRewriter
         public override BoundNode VisitHostObjectMemberReference(BoundHostObjectMemberReference node)
         {
             Debug.Assert(_previousSubmissionFields != null);
-            Debug.Assert(!_factory.CurrentMethod.IsStatic);
+            Debug.Assert(!_factory.CurrentMethod.GetContainingMember().IsStatic);
```

Comment:
Would `_factory. TopLevelMethod` do the job here (and in the other modified assert)?
If so, then the `GetContainingMember` extension method can be inlined back into `ContainingMember()` (in `Binder_Expression.cs`).
Thanks @AlekseyTs for the tip